### PR TITLE
chore: Deprecate extra in favor of driver options

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
Deprecate databse_extra in favor of database_driver_options

### CODE REVIEW

File: packages/types/src/common/config-module.ts [hunk 1]
```diff
@@ -342,6 +342,8 @@ export type ProjectConfigOptions = {
    *   // ...
    * }
    * ```
+   *
+   * @deprecated use `database_driver_options` instead
```
Comment (olivermrbl) [inline packages/types/src/common/config-module.ts]: question: For Typeorm connections, should this be backward compatible?

Comment (adrien2p) [reply packages/types/src/common/config-module.ts]: The typeorm datasource is not loaded if the v2 flag is enabled, so in theory this works only for v1 right?

Comment (olivermrbl) [reply packages/types/src/common/config-module.ts]: My thinking was that if we release this tomorrow, `database_extra` will be denoted as deprecated, and v1 users will attempt to use `database_driver_options` instead, which is incompatible with the Typeorm datasource. I might be overthinking here, let me know if it makes sense.

Comment (srindom) [reply packages/types/src/common/config-module.ts]: think it's fine to leave out the deprecation warning for now - when we get to v2 we should probably remove it altogether

Comment (adrien2p) [reply packages/types/src/common/config-module.ts]: ahah you are right sorry, I ll remove that flag

**Comment by adrien2p** [comment]
@srindom should we slap an automerge on this one?

**Review by srindom — APPROVED** [review]

**Pull request merged**